### ISSUE TITLE
[Accessibility] Opening the console can require multiple <escape> presses

### ISSUE THREAD

**Issue by jasonLaster** [devtools-html/debugger.html#1849]
When CodeMirror is in focus, it requires two escape presses to open the console.

1. `escape` -> stop focusing on codemirror
2. `escape` -> open console

@npsecoding I think is related to https://github.com/devtools-html/debugger.html/commit/71360324aed48f8978ef00520596b7f40c46fb72

![](http://g.recordit.co/IfHRYIhVNK.gif)

**Comment by npsecoding** [comment]
Hey Jason.

This is the expected behavior for accessibility purposes, to give user's a way to navigate out of the code editor using the keyboard. The first escape focuses on the editor container and the second escape results in the default action of opening the console. If you press escape on any other point of the debugger the console should be brought up immediately.

**Comment by Garbee** [comment]
Yah. `esc` pulls you out of the editor but keeps the tab placement where it is at. So you can tab to the next area in the content flow if desired from that point. The second `esc` is perfectly fine in this scenario. Quirky to the majority that don't navigate with a keyboard, but expected for those that do.

**Comment by juliandescottes** [comment]
I think that's one of the things I noticed when we upgraded the bundle in mc.

My main concerns about this were:
- after pressing escape once, the focus is on the editor container, which we can see with the dotted outline. Pressing TAB puts the focus back in the editor, so there's no way to actually get past the editor using keyboard navigation. Pressing shift+TAB does focus the previous element in the flow, but we don't have :focus outlines for those so it's easy to misunderstand.
- this is a behaviour change compared with the previous version of the bundle, or compared with the old debugger, or the chrome debugger.

**Comment by Garbee** [comment]
Hmm, didn't catch the focus going back into the editor. We should probably detect `esc` on the editor, and when done shift focus to the next focusable node? Need to do some research here to figure out the best way forward.

Chrome's tools can't really be checked against for a11y concerns as they don't support much keyboard navigation in their sources panel (and possibly other areas as well.)

+1 to investigating alternative shortcuts to not collide with more natural keyboard navigation patterns.

**Comment by jasonLaster** [comment]
Thanks guys. It sounds like:

a) we'd like maintain native keyboard actions i.e. <escape> moves the focus out of the editor
b) we'd like a better focus state for the editor
c) we *are* deviating from previous behavior where we used to treat escape as a global shortcut to open the console.

I'm going to close this issue because it seems like we're not going to change the behavior. Feel free to follow up though.
